airflow/providers/ssh/operators/ssh.py

```diff
@@ -198,3 +198,9 @@ def tunnel(self) -> None:
         """Get ssh tunnel."""
         ssh_client = self.hook.get_conn()  # type: ignore[union-attr]
         ssh_client.get_transport()
+
+    def on_kill(self) -> None:
+        """Close the ssh client session."""
+        ssh_client = self.hook.get_conn()
```

Comment:
I think we also need to add  caching to `get_conn` - otherwise every call to get_conn() retrieves a new client

Reply:
This is how get_conn() look like now:

```python
    def get_conn(self) -> paramiko.SSHClient:
        ....
        client = paramiko.SSHClient()
        ...
        self.client = client
        return client
```

Reply:
Yeah, I see. I added a `cached_property` decorator to `get_conn` based on the cached propert `host_proxy` in `SSHHook`. From what I can see, this shouldn't not affect existing code.

Reply:
It's not the same - it turns `get_conn()` into `get_conn` property which is not what we want - it won't change caching when you directly call get_conn() method, which is what we do in a few places.

Look at other hooks (find `def get_conn(` in other files - we usually do `if self.client: return self.client` kind of approach there.